### ISSUE TITLE
`:dumptests` doesn't print one test case per line

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#1314]
The REPL documentation for `:? :dumptests` suggests that "each line" of the resulting file should contain the inputs and expected output for one random test. However, since version 2.12.0, values of larger types (arrays or records) are pretty-printed with lots of extra line breaks, so each test case covers many lines. For example:
```
Cryptol> :set tests=3
Cryptol> :dumptests foo.txt \(xs:[4][32]) -> xs == xs
```
This produces the following 12-line file:
```
True    [0x63ed9cea,
0xb442e53d,
0xda7f35e6,
0xa5221c2d]
True    [0x39b0818b,
0x1d79967e,
0xed26edc2,
0x2314627f]
True    [0x407dd58e,
0x3d12445c,
0x29f39490,
0xd4086e5d]
```

**Comment by brianhuffman** [comment]
Bisection shows that PR #1263 is where the problem originates.

**Comment by brianhuffman** [comment]
The problem is in the definition of the `renderOneLine` function in `Utils/PP.hs`: https://github.com/GaloisInc/cryptol/blob/07c689a699a7e2508da51bb22f63b0d3667ac15d/src/Cryptol/Utils/PP.hs#L141-L142
The `layoutCompact` style in `prettyprinter` doesn't do single-line layouts; it just skips indentation. We'll need to implement this another way.

**Comment by robdockins** [comment]
I think maybe we can implement this by writing a custom rendering function that never emits any newlines.  It should be easier than changing the pretty printing library.

**Comment by brianhuffman** [comment]
It would probably be worth asking the `prettyprinter` devs if there is a recommended way to do one-line layouts.

**Comment by kquick** [comment]
Potentially useful example: https://hackage.haskell.org/package/silkscreen-0.0.0.4

**Comment by RyanGlScott** [comment]
Fixed in #1759.
